### PR TITLE
add git secrets ci ros2

### PR DESCRIPTION
**Pull request by ojasjoshi**
*Issue #, if available:*

*Description of changes:*
Adding git secrets scan action to ROS2 workflow

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ojasjoshi** [comment]
Blocked by aws-robotics/aws-robomaker-github-actions#17.

Before merging, requires version bump (to 2.3.4) for all aws-robomaker-gh-actions after is merged.

File: .github/workflows/ros2.yml [hunk 1]
```diff
@@ -32,6 +32,8 @@ jobs:
       run: |
         # TODO(ros-tooling/setup-ros-docker#7): calling chown is necessary for now
         sudo chown -R rosbuild:rosbuild "$HOME" .
+    - name: test-git-secret-actions
+      uses: aws-robotics/aws-robomaker-github-actions/git-secrets-scan-action@2.3.3
```
Comment (mm318) [inline .github/workflows/ros2.yml]: We'll need to use a newer version of the `aws-robomaker-github-actions` GitHub repo.

File: .github/workflows/ros2.yml [hunk 2]
```diff
@@ -32,6 +32,8 @@ jobs:
       run: |
         # TODO(ros-tooling/setup-ros-docker#7): calling chown is necessary for now
         sudo chown -R rosbuild:rosbuild "$HOME" .
+    - name: test-git-secret-actions
```
Comment (mm318) [inline .github/workflows/ros2.yml]: The naming convention here should be something like "Git Secrets Scan".

**Comment by mm318** [comment]
> Before merging, requires version bump (to 2.3.4) for all aws-robomaker-gh-actions after is merged.

I think it should be a bigger bump, like to `2.4.0` since it's adding a new action/feature (without breaking backward compatibility).

**Review by mm318 — APPROVED** [review]

**Pull request merged**